### PR TITLE
Test that profiles memory use for programs with lateness

### PR DESCRIPTION
**Pull request by mihaibudiu**
Is this a user-visible change (yes/no): no

### CODE REVIEW

File: sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/streaming/StreamingTests.java [hunk 1]
```diff
@@ -96,6 +108,117 @@ CREATE TABLE series (
...
+                   let profile = circuit.retrieve_profile().expect("could not get profile");
+                   let end = SystemTime::now();
+                   let duration = end.duration_since(start).expect("could not get time");
+
+                   let mut data = String::new();
+                   data.push_str(&format!("{},{}\\n",
+                                          duration.as_millis(),
+                                          profile.total_used_bytes().unwrap().bytes));
+                   let mut file = File::create("mem.txt").expect("Could not create file");
+                   file.write_all(data.as_bytes()).expect("Could not write data");
```
Comment (blp) [inline sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/streaming/StreamingTests.java:164]: You could use [`std::fs::write`](https://doc.rust-lang.org/stable/std/fs/fn.write.html) here.

**Review by blp — APPROVED** [review]
I don't think I have any special insight into this commit. Is there anything particular I should look for?

I added one comment, it's just a nitpick, feel free to ignore it.

File: sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/streaming/StreamingTests.java (hunk 1 above)
Comment (mihaibudiu) [reply sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/streaming/StreamingTests.java:164]: Well, it's mostly Rust, so I thought to bother you. I rotate Rust reviewers.
Thanks.

Comment (mihaibudiu) [reply sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/streaming/StreamingTests.java:164]: But I expect that at some point we will built a monitoring agent for pipelines based on a similar skeleton, exposing virtual views with program statistics.

**Pull request merged**